### PR TITLE
Migrate to kfp 2

### PR DESCRIPTION
**Pull request by sfczekalski**
#### Description

Migrated kfp from legacy 1.X.X to 2.8.0.

##### PR Checklist
- [x] Tests added
- [x] [Changelog](CHANGELOG.md) updated

### CODE REVIEW

**Comment by em-pe** [comment]
Good work, just address the issues that sonarcloud caught https://sonarcloud.io/project/issues?issueStatuses=OPEN%2CCONFIRMED&pullRequest=170&id=getindata_kedro-vertexai

Additionally maybe we should update the python version that runs e2e tests to python 3.10 in https://github.com/getindata/kedro-vertexai/blob/56fff64eb34a3250919f066d32214e033ee28ff3/.github/workflows/test_and_publish.yml#L108 and https://github.com/getindata/kedro-vertexai/blob/56fff64eb34a3250919f066d32214e033ee28ff3/.github/workflows/test_and_publish.yml#L194

File: kedro_vertexai/cli.py [hunk 1]
```diff
@@ -98,14 +97,6 @@ def list_pipelines(ctx):
     help="Parameters override in form of `key=value`",
 )
 @click.option("--wait-for-completion", type=bool, is_flag=True, default=False)
-@click.option(
```
Comment (adobrogows-ki) [inline kedro_vertexai/cli.py:101]: Why was timeout removed?

File: pyproject.toml [hunk 2]
```diff
@@ -35,7 +35,7 @@ known_third_party = ["click","google","kedro","kfp","kubernetes","tabulate", "py
 python = ">=3.8,<3.11" # weird error OverrideNeeded when bumping up. Could be related to old versions with kfp
 kedro = ">=0.19.0,<0.20"
 click = ">=8.0.4"
-kfp = ">=1.8.21,<2.0"
+kfp = "2.8.0"
```
Comment (adobrogows-ki) [inline pyproject.toml:38]: can this requirement be softened? It's not okay for secondary libraries to make very strict version restrictions

File: kedro_vertexai/cli.py (hunk 1 above)
Comment (sfczekalski) [reply kedro_vertexai/cli.py:101]: In the old version of the plugin, the timeout meant only the timeout of the local python process (how long to wait for the pipeline completion until the execution of the local command was exited), not the timeout of the actual remote pipeline on Vertex AI. So even if the pipeline execution time exceeded the specified timeout value, the remote pipeline won't be altered in any way and will continue to run.
I've talked about it with @em-pe and the plan is to develop a proper timeout that would also alter the remote pipeline execution, possibly including the per-task timeout, not only the whole pipeline timeout.

File: pyproject.toml (hunk 2 above)
Comment (sfczekalski) [reply pyproject.toml:38]: Good point, thank you! I'll soften this requirement, possibly just `kfp > "2.0"` wdyt?

**Review by adobrogows-ki — COMMENTED** [review]
Added 2 questions, I didn't catch anything that would need fixing. Good job :) I'm not familiar with kfp that much and would need to run it myself to check how it works now.

File: kedro_vertexai/cli.py (hunk 1 above)
Comment (adobrogows-ki) [reply kedro_vertexai/cli.py:101]: Ok, some of this info should be included in the changelog as it's an api change of usage of our plugin and should be documented. Probably docs also need updating.

File: pyproject.toml (hunk 2 above)
Comment (adobrogows-ki) [reply pyproject.toml:38]: Yes, I think it's fine, >= 2.0. Let's maybe also test for 2.0 version to be make sure nothing breaks between those versions.

File: kedro_vertexai/cli.py (hunk 1 above)
Comment (sfczekalski) [reply kedro_vertexai/cli.py:101]: Ok, sure!

File: pyproject.toml (hunk 2 above)
Comment (sfczekalski) [reply pyproject.toml:38]: Ok, I'll do that!

Comment (em-pe) [reply pyproject.toml:38]: `>=2.0, <3`

File: tests/e2e/standard/overwrite/conf/base/vertexai.yml [hunk 3]
```diff
@@ -4,10 +4,6 @@ run_config:
   # Name of the image to run as the pipeline steps
   image: gcr.io/gid-ml-ops-sandbox/kedro-vertexai-e2e:standard
 
-  # Pull policy to be used for the steps. Use Always if you push the images
-  # on the same tag, or Never if you use only local images
-  image_pull_policy: IfNotPresent
```
[the remark this answers is not in the document]
Comment (sfczekalski) [reply tests/e2e/standard/overwrite/conf/base/vertexai.yml:9]: @em-pe actually, as I understand, the `image_pull_policy` argument is not applicable anymore - it doesn't exist in `kfp` 2, and it only exists in `kfp-kubernetes` extension package and can be used when we run the pipelines on Kubernetes. Not sure how Vertex AI works under the hood, I've tried to find information how it approaches fetching the images, whenever it uses some caching or not, but I couldn't find this piece of information...

**Review by em-pe — APPROVED** [review]

**Pull request merged**